scripts/manage-build-cache.py

```
@@ -0,0 +1,353 @@
+def immediate_entries(path: Path) -> list[dict[str, Any]]:
+    entries = []
+    if path.is_dir():
+        for child in path.iterdir():
+            size, newest = tree_metrics(child)
+            entries.append({"path": str(child), "bytes": size, "newest_mtime": newest})
+    return sorted(entries, key=lambda entry: entry["bytes"], reverse=True)
+
+
+def cargo_metadata(workspace: Path) -> dict[str, Any]:
+    result = subprocess.run(
+        ["cargo", "metadata", "--no-deps", "--format-version", "1"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)